Fix/ecp 1061 default organizer state

stratease opened the pull request.

EugeneKyale commented on src/modules/blocks/event-organizer/container.js:
```diff
@@ -1,6 +1,7 @@
+import React, { useEffect } from 'react';
 import { compose } from 'redux';
 import { connect } from 'react-redux';
 import { uniq } from 'lodash';
```
```suggestion
import { uniq } from 'lodash';
import PropTypes from 'prop-types';
```

EugeneKyale commented on src/modules/blocks/event-organizer/container.js:
```diff
@@ -99,10 +102,26 @@ const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
+		if ( ! organizer && defaults && defaults.organizer ) {
+			props.setAttributes( { organizer: defaults.organizer } );
+		}
+	}, [] )
```
```suggestion
	}, [ props ] );
```

stratease replied:
@EugeneKyale don't think we want this to re-run. It just loads defaults on initial run if we have a default available and if there isn't an existing organizer already set (assuming we will always have the `organizer` prop set on first render). Right?

EugeneKyale commented on src/modules/blocks/event-organizer/container.js:
```diff
@@ -99,10 +102,26 @@ const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
+
+	return (
+		<EventOrganizer { ...props } />
+	)
```
```suggestion
	);
```

EugeneKyale commented on src/modules/blocks/event-organizer/container.js:
```diff
@@ -99,10 +102,26 @@ const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
+	return (
+		<EventOrganizer { ...props } />
+	)
+}
```
```suggestion
};

StatefulEventOrganizer.propTypes = {
	attributes: PropTypes.array,
	setAttributes: PropTypes.func,
};
```

EugeneKyale commented on src/modules/blocks/event-organizer/container.js:
```diff
@@ -99,10 +102,26 @@ const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
 	withDetails( 'organizer' ),
 	connect( mapStateToProps, mapDispatchToProps, mergeProps ),
-)( EventOrganizer );
+)( StatefulEventOrganizer );
```
```suggestion
)( StatefulEventOrganizer );

```

EugeneKyale reviewed (commented):
The approach looks good to me. I just a few small comments.

bordoni commented on src/modules/blocks/event-organizer/container.js:
```diff
@@ -99,10 +103,33 @@ const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
 	};
 };
 
+const StatefulEventOrganizer = ( props ) => {
+	// This hook should only run once, it checks for default values.
+	useEffect( () => {
+		// Manage our initial state for defaults.
+		const defaults = editorDefaults();
+		const { attributes: { organizer } } = props;
+
+		if ( organizer === null && defaults && defaults.organizer ) {
+			props.setAttributes( { organizer: defaults.organizer } );
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	return (
+		<EventOrganizer { ...props } />
+	);
+};
```
Docblock.

stratease commented:
> I'll approve despite the failing lint tests because I believe they're part of a broader issue.

Yeah, I fixed the ones I saw that were of the files I modified. The others are existing lint failures - figure we can get them over time.
